### PR TITLE
KAFKA-4406: add ssl.provider.classes config option

### PR DESCRIPTION
**Pull request by reftel**

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/common/security/ssl/SslFactory.java [hunk 1]
```diff
@@ -123,7 +133,12 @@ else if (clientAuthConfig.equals("requested"))
     }
 
 
-    private SSLContext createSSLContext() throws GeneralSecurityException, IOException  {
+    private SSLContext createSSLContext()
+        throws GeneralSecurityException, IOException, ClassNotFoundException, IllegalAccessException,
+        InstantiationException {
+        for (String providerClassName : providerClasses) {
+            Security.addProvider((Provider) Class.forName(providerClassName).newInstance());
```
Comment (harshach) [inline clients/src/main/java/org/apache/kafka/common/security/ssl/SslFactory.java]: can you use the Utils.newInstance here . https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/utils/Utils.java#L312

**Review by harshach — COMMENTED** [review]
overall looks good to me.

**Comment by reftel** [comment]
Thanks for your comments! I've updated as you suggested, and also fixed some issues found by checkstyle etc.
Please note, though, that this change is being discussed at https://issues.apache.org/jira/browse/KAFKA-4406 , and should not be merged until the discussion has concluded.

**Comment by omkreddy** [comment]
This was implemented in KAFKA-8669

**Pull request closed without merging**